crates/bevy_render/src/render_task/compute_builder.rs

```diff
@@ -0,0 +1,185 @@
+            push_constants: None,
+            bind_groups: Vec::new(),
+            bind_group_layouts: Vec::new(),
+            resource_cache,
+            pipeline_compiler,
+            render_device,
+        }
+    }
+
+    pub fn shader(mut self, shader: Handle<Shader>) -> Self {
+        self.shader = shader;
+        self
+    }
+
+    pub fn entry_point(mut self, entry_point: &'static str) -> Self {
+        self.entry_point = Some(entry_point);
+        self
+    }
+
+    pub fn shader_def(mut self, shader_def: impl Into<ShaderDefVal>) -> Self {
+        self.shader_defs.push(shader_def.into());
+        self
+    }
+
+    pub fn shader_def_if(mut self, shader_def: impl Into<ShaderDefVal>, condition: bool) -> Self {
+        if condition {
+            self.shader_defs.push(shader_def.into());
+        }
+        self
+    }
+
+    pub fn push_constants<T: NoUninit>(mut self, push_constants: &'a [T]) -> Self {
+        self.push_constants = Some(bytemuck::cast_slice(push_constants));
+        self
+    }
+
+    pub fn bind_resources<'b, const N: usize>(
+        mut self,
+        resources: impl IntoBindingArray<'b, N> + IntoBindGroupLayoutEntryBuilderArray<N> + Clone,
+    ) -> Self {
+        let layout_descriptor = BindGroupLayoutDescriptor::new(
+            self.pass_name.to_owned(),
+            &BindGroupLayoutEntries::sequential(ShaderStages::COMPUTE, resources.clone()),
+        );
+
+        let descriptor = BindGroupDescriptor {
+            label: Some(self.pass_name),
+            layout: &self
+                .pipeline_compiler
+                .get_bind_group_layout(&layout_descriptor),
+            entries: &BindGroupEntries::sequential(resources),
+        };
+
+        // TODO
+        // self.bind_groups.push(Some(
+        //     self.resource_cache
+        //         .get_or_create_bind_group(descriptor, self.render_device),
+        // ));
+        self.bind_groups.push(Some(
+            self.render_device
+                .wgpu_device()
+                .create_bind_group(&descriptor)
+                .into(),
+        ));
+
+        self.bind_group_layouts.push(layout_descriptor);
+
+        self
+    }
+
+    pub fn bind_group(
+        mut self,
+        bind_group: impl Into<Option<BindGroup>>,
+        layout: BindGroupLayoutDescriptor,
+    ) -> Self {
+        self.bind_groups.push(bind_group.into());
+        self.bind_group_layouts.push(layout);
+        self
+    }
+
+    pub fn dispatch_1d(mut self, x: u32) -> Option<Self> {
+        self.setup_state()?;
+        self.pass.dispatch_workgroups(x, 1, 1);
+        Some(self)
+    }
+
+    pub fn dispatch_2d(mut self, x: u32, y: u32) -> Option<Self> {
+        self.setup_state()?;
+        self.pass.dispatch_workgroups(x, y, 1);
+        Some(self)
+    }
+
+    pub fn dispatch_3d(mut self, x: u32, y: u32, z: u32) -> Option<Self> {
+        self.setup_state()?;
+        self.pass.dispatch_workgroups(x, y, z);
+        Some(self)
+    }
```

Comment:
Is there much value in these three instead of just `dispatch_3d` called `dispatch`?